src/l2/eth.rs

```diff
@@ -0,0 +1,28 @@
+use core::mem;
+
+pub const ETH_HDR_LEN: usize = mem::size_of::<EthHdr>();
+
+/// Ethernet header, which is present at the beginning of every Ethernet frame.
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone)]
+pub struct EthHdr {
+    /// Destination MAC address.
+    pub dest: [u8; 6],
```

Comment:
Either go with `source` and `destination` or `src` and `dst`.
I would also recommend that you use something like `src_hwaddr` and `dst_hwaddr` and/or `src_addr` and `dst_addr` so these are clearly mac/ip addresses and not ports.
It might seem an arbitrary distinction here, but imagine in more complex code with poor variable names like this:
```rust
let e = EthHdr::new();
let i = Ipv4Hdr::new();
let t = TcpHdr::new();

info!("source_mac: {}, source_ip: {}, source_port: {}", e.src, i.src, t.src);
```